promql/engine.go

```diff
@@ -668,16 +689,21 @@ func (enh *EvalNodeHelper) signatureFunc(on bool, names ...string) func(labels.L
 
 // rangeEval evaluates the given expressions, and then for each step calls
 // the given function with the values computed for each expression at that
-// step.  The return value is the combination into time series of  of all the
+// step.  The return value is the combination into time series of all the
 // function call results.
-func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ...Expr) Matrix {
+func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ...Expr) (Matrix, error) {
```

Comment:
You shouldn't need to change this signature, use ev.error

Reply:
we want to panic and recover all error cases? that seems strange to me

Reply:
Yes, that's how error handling is done in PromQL.